### PR TITLE
Scrape BIKI for functions

### PR DESCRIPTION
**Pull request by Cuel**
Scrapes BIKI for functions, outputs json to /settings folder

Instructions -> navigate to folder, do "npm install" > "node ."

### CODE REVIEW

**Comment by thojkooi** [comment]
:+1: Thanks @Cuel

**Comment by bux** [comment]
:+1: for es6

File: rsc/retrieve-bi-fncs/index.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  root.find('h3').each(function () {
+    $(this).next().find('li a[href]').each(function () {
+      let text = $(this).attr('title').trim().replace(/\s/g, '_')
+      let descriptionMoreURL = URL_BASE + $(this).attr('href').trim()
+      ret.push({
+        text,
+        descriptionMoreURL,
+        rightLabel: 'BIS Function',
+        type: 'function',
+        description: ''
```
Comment (jonpas) [inline rsc/retrieve-bi-fncs/index.js]: Change to same order as CBA functions, just to keep a standard.
- Text
- Right Label
- Type
- Description
- Description More URL

File: rsc/retrieve-bi-fncs/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "bi-fnc-scraper",
```
Comment (jonpas) [inline rsc/retrieve-bi-fncs/package.json]: `bis-fnc-scraper` to keep it the same.

File: rsc/retrieve-bi-fncs/index.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+'use strict'
+const write = require('fs').writeFileSync
+const normalize = require('path').normalize
+const request = require('request')
+const cheerio = require('cheerio')
+const chalk = require('chalk')
+
+const URL_BASE = 'http://community.bistudio.com'
+const URL_FNC = `${URL_BASE}/wiki/Category:Arma_3:_Functions`
+const OUTPUT_FILE = normalize(`${__dirname}/../../settings/language-sqf-functions-bis.json`)
```
Comment (jonpas) [inline rsc/retrieve-bi-fncs/index.js]: I'd rather use `BI`.

**Comment by jonpas** [comment]
Can `BIS`/`BI` be streamlined? To whatever you prefer (officially it's just BI afaik, but they name their functions BIS so whatever).

Otherwise it seems to work nicely, good job! :+1:

File: rsc/retrieve-bi-fncs/index.js (hunk 1 above)
Comment (Cuel) [inline rsc/retrieve-bi-fncs/index.js]: Sure, key order is not normally guaranteed but V8 seems to match the insertion order at least

**Comment by jonpas** [comment]
Perfect, thank you!

Merging.

**Pull request merged**